### ISSUE TITLE
Weird `changeUser()` behavior

### ISSUE THREAD

**Issue by matinzd** [Appboy/appboy-react-sdk#139]
Hi,
As you can find this bug [here](https://github.com/Appboy/appboy-android-sdk/blob/64774c3c64b5481e758de8f518ef56bdbdf4432d/CHANGELOG.md#added), if you try to call `changeUser()` with wrong SDK auth token and new user id, it won't do anything anymore. This fix is available on your version 18.0.0 android SDK. I think you need to update react-native SDK as well. 

https://github.com/Appboy/appboy-android-sdk/blob/64774c3c64b5481e758de8f518ef56bdbdf4432d/CHANGELOG.md#added 
```
Added
Added the ability to change SDK authentication signature with Braze.changeUser() when the current user id and a new signature is passed in.
Previously, Braze.changeUser() would not change the SDK authentication signature if the current user id was used.
```

**Comment by radixdev** [comment]
Hi @matinzd ,

Can you elaborate on the issue you're seeing and concrete steps to reproduce?

**Comment by matinzd** [comment]
> Hi @matinzd ,
> 
> Can you elaborate on the issue you're seeing and concrete steps to reproduce?

First of all this issue is android-specific:
- First try to changeUser with correct user id and auth token
- Then try to changeUser with a different new user id but same auth token (Then your server will throw sdk auth error)
- After that, try to call changeUser with new user id but correct auth token this time.
- Check adb logs
- It says changing user to the same user id, doing nothing. In this case android sdk should change the auth token but not user id. This causes the sdk not to update the auth token so the new user will not be registered.

**Comment by Bucimis** [comment]
@matinzd an update that contains Android 18.0.1 will be released shortly

**Comment by davidbielik** [comment]
Hi @matinzd as a workaround you can use the `setSdkAuthenticationSignature` method: https://www.braze.com/docs/developer_guide/platform_wide/sdk_authentication/#set-the-current-users-jwt-token

**Comment by matinzd** [comment]
Ok but please resolve this when the new version is out.

**Comment by davidbielik** [comment]
Hi @matinzd we've just released v1.3.4 which upgrades to the latest Android SDK. Hope this helps!

**Comment by matinzd** [comment]
@davidbielik Thanks!
